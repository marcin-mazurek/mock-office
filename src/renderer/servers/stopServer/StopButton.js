import React from 'react';
import { connect } from 'react-redux';
import init from './actions';
import { getSelected } from '../selectors';

export const StopButton = ({ serverId, stop }) =>
  <button
    className="button server-stop-button"
    style={{ backgroundColor: '#AAA', color: '#FFF', padding: '5px 10px' }}
    onClick={() => {
      stop(serverId);
    }}
  >
    Stop
  </button>;

StopButton.propTypes = {
  serverId: React.PropTypes.string.isRequired,
  stop: React.PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  serverId: getSelected(state)
});

const mapDispatchToProps = {
  stop: init
};

export default connect(mapStateToProps, mapDispatchToProps)(StopButton);
