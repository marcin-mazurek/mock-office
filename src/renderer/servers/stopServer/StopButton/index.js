import React from 'react';
import { connect } from 'react-redux';
import { Button } from 'react-mdl';
import init from '../actions';
import { getSelected } from '../../selectors';

const StopButton = ({ serverId, stop }) => (
  <Button
    raised
    ripple
    style={{ backgroundColor: '#ff5959', color: '#FFF' }}
    onClick={() => {
      stop(serverId);
    }}
  >
    {'Stop'}
  </Button>
);

StopButton.propTypes = {
  serverId: React.PropTypes.string.isRequired,
  stop: React.PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  serverId: getSelected(state)
});

const mapDispatchToProps = {
  stop: init
};

export default connect(mapStateToProps, mapDispatchToProps)(StopButton);
