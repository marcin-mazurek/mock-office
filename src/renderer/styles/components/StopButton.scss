$size: 50px;

.server-stop-button {
  font-size: 23px;
  border-radius: 100%;
  cursor: pointer;
  height: $size;
  width: $size;
  background-color: $stop-button-background-color;
  color: $stop-button-font-color;
  display: flex;
  align-items: center;
  justify-content: center;
  border: $stop-button-border;
}
