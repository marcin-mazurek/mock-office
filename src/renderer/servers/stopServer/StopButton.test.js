import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { StopButton } from './StopButton';

describe('StopButton', () => {
  test('default snapshot', () => {
    const wrapper = shallow(<StopButton serverId="some id" stop={() => {}} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
