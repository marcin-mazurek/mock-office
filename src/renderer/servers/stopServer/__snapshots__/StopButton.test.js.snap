exports[`test StopButton should render 1`] = `
<button
  className="button server-stop-button"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#AAA",
      "color": "#FFF",
      "padding": "5px 10px",
    }
  }>
  Stop
</button>
`;
